sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -221,63 +219,82 @@ func _main() int {
 		log.G(ctx).WithError(err).WithField("img", img.Name).Error("Failed to create container")
 		return 1
 	}
-	defer container.Delete(context.TODO(), containerd.WithSnapshotCleanup)
+	defer func() {
+		// Clean up the container as program wraps up.
+		cleanup, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+		err := container.Delete(cleanup, containerd.WithSnapshotCleanup)
+		if err != nil {
+			log.G(cleanup).WithError(err).Error("Failed to cleanup container")
+		}
+	}()
 
 	// Create the container task
 	task, err := container.NewTask(ctx, cio.NewCreator(cio.WithStdio))
 	if err != nil {
 		log.G(ctx).WithError(err).Error("Failed to create container task")
 		return 1
 	}
-	defer task.Delete(context.TODO())
+	defer func() {
+		// Clean up the container's task as program wraps up.
+		cleanup, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+		_, err := task.Delete(cleanup)
+		if err != nil {
+			log.G(cleanup).WithError(err).Error("Failed to delete container task")
+		}
+	}()
 
-	// Wait before calling start in case the container task finishes too quickly
+	// Wait to call start in case the container task finishes too quickly.
```

Comment:
The previous comment is actually more correct.  `task.Wait` is roughly patterned after `wait(2)` in that you'll receive notification when the task exits.  If the task exits before you call `Wait`, you don't get to read its status.  By calling `Wait` before `Start`, you guarantee that you receive the notification.

We're not delaying the start, but monitoring.

Reply:
Ah, yep I read this, as you pointed out, saying it is a *delayed* call - I'll fix the comment.